packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -161,178 +155,145 @@ describe('src/cy/commands/cookies - no stub', () => {
       cy.getCookie('foo', { domain: 'www.foobar.com' })
       .its('domain').should('match', /\.?www\.foobar\.com/)
 
-      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+      // domain is exact
+      cy.getCookie('key', { domain: 'foobar.com' }).should('be.null')
+
+      // webkit does not support cy.origin()
+      if (isWebkit) return
 
       cy.origin('http://www.foobar.com:3500', () => {
         cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
         cy.setCookie('foo', 'bar', { domain: 'barbaz.com' })
 
         cy.getCookie('foo', { domain: 'barbaz.com' })
         .its('domain').should('match', /\.?barbaz\.com/)
+
+        // domain is exact
+        cy.getCookie('key', { domain: 'barbaz.com' }).should('be.null')
       })
     })
   })
 
   context('#clearCookies', () => {
-    it('clears all cookies', () => {
-      cy.setCookie('foo', 'bar')
-      cy.getCookies().should('have.length', 1)
-      cy.clearCookies()
-      cy.getCookies().should('have.length', 0)
+    beforeEach(() => {
+      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
+      cy.setCookie('key1', 'value1')
+      cy.setCookie('key2', 'value2')
+      cy.setCookie('key3', 'value3', { domain: 'barbaz.com' })
+      cy.setCookie('key4', 'value4', { domain: 'foobar.com' })
+      cy.setCookie('key5', 'value5', { domain: 'www.foobar.com' })
+      cy.setCookie('key6', 'value6', { domain: 'www.foobar.com' })
     })
 
     it('clears the cookies on the domain matching the AUT by default', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar')
-      cy.setCookie('baz', 'qux')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-
       cy.clearCookies()
 
-      cy.getCookie('foo').should('be.null')
-      cy.getCookie('baz').should('be.null')
-      cy.getCookie('foo', { domain: 'www.foobar.com' }).should('exist')
+      cy.getCookie('key1').should('be.null')
+      cy.getCookie('key2').should('be.null')
+      cy.getCookie('key3', { domain: 'barbaz.com' }).should('exist')
```

Comment:
why didn't this one clear?

Reply:
Right, that's the primary change in this PR. The domain must be exact.